src/js/main.js

```diff
@@ -42,24 +42,58 @@ function translate(text, sourceLang, targetLang, callback) {
 
   console.log('translate', sourceLang, targetLang);
 
-  var params = {
-    target: targetLang,
-    q: text
-  };
+  $.get(chrome.extension.getURL('api-proxy-sign'), function(signature) {
+    $.ajax({
+      url: 'https://web-api-proxy.herokuapp.com/language/translate/v2/detect',
```

Comment:
First we detect the language so we know whether we are translating from English (in which case the user will be selecting the target language instead of the source language).